db/user_test.go

```diff
@@ -14,6 +17,162 @@ type SyncNewUsersTestCase struct {
+            []string{"a@test.com"},
+            nil,
+            []string{},
+            0,
+        },
+
+        // This is a basic test to ensure that a role gets mapped to the correct email.
+        {
+            []string{"admin"},
+            []string{"admin@test.com"},
+            nil,
+            []string{},
+            0,
+        },
+
+        // This is a test for our all roles character, the *. 
+        {
+            []string{"*"},
+            []string{"admin@test.com", "grader@test.com", "other@test.com", "owner@test.com", "student@test.com"},
+            nil,
+            []string{},
+            0,
+        },
+
+
+        // This test case is given redundant roles and emails.
+        // It tests to ensures we do not produce duplicates on this input.
+        {
+            []string{"other", "*", "grader@test.com"},
+            []string{"admin@test.com", "grader@test.com", "other@test.com", "owner@test.com", "student@test.com"},
+            nil,
+            []string{},
+            0,
+        },
+
+        // This test case tests if miscapitalized roles still function.
+        {
+            []string{"OTHER"},
+            []string{"other@test.com"},
+            nil,
+            []string{},
+            0,
+        },
+
+        // This test case tests if warnings are issued on invalid roles.
+        {
+            []string{"trash", "garbage", "waste", "recycle!"},
+            []string{},
+            nil,
+            []string{},
+            4,
+        },
+
+        // This test adds new Users to the course and ensures we retrieve all emails for the given role.
+        {
+            []string{"student"},
+            []string{"a_student@test.com", "b_student@test.com", "student@test.com"},
+            []*model.User{model.NewUser("a_student@test.com", "", model.GetRole("student")), model.NewUser("b_student@test.com", "", model.GetRole("student"))},
+            []string{},
+            0,
+        },
+
+        // This is a test case to see if we properly trim whitespace.
+        {
+            []string{"\t\n student    ", "\n \t testing@test.com", "\t\n     \t    \n"},
+            []string{"student@test.com", "testing@test.com"},
+            nil,
+            []string{},
+            0,
+        },
+
+        // This test case removes the only user from the "owner" role, so we check that a role without any users still functions properly.
+        {
+            []string{"owner", "student"},
+            []string{"student@test.com"},
+            nil,
+            []string{"owner@test.com"},
+            0,
+        },
+    };
+
+    for i, testCase := range testCases {
+        ResetForTesting();
+        course := MustGetCourse(TEST_COURSE_ID);
+        // startTime := time.Time{};
+
+        for _, newUser := range testCase.addUsers {
+            SaveUser(course, newUser);
+        }
+
+        for _, oldUser := range testCase.removeUsers {
```

Comment:
"oldUser" is not a good name (since the user is not old).
(I see the symmetry with "newUser, but we are using a different meaning of "new" here.)

"removeUser"?